### PR TITLE
fix RandomSampler length

### PR DESCRIPTION
**Pull request by truskovskiyk**
Hi!

This PR addresses #15537  issue.
Please review.

Thanks!

cc: @fmassa @jbaron @soumith

### CODE REVIEW

**Review by jbaron — APPROVED** [review]
The correct patch as discussed in the issue: the dataset now returns the correct length in case num_samples is specified. Also good that there are now some tests in place which prevent a regression issue to popup in the future. Thanks!

File: torch/utils/data/sampler.py [hunk 1]
```diff
@@ -50,30 +50,34 @@ class RandomSampler(Sampler):
...
         if not isinstance(self.num_samples, int) or self.num_samples <= 0:
             raise ValueError("num_samples should be a positive integeral "
                              "value, but got num_samples={}".format(self.num_samples))
         if not isinstance(self.replacement, bool):
             raise ValueError("replacement should be a boolean value, but got "
                              "replacement={}".format(self.replacement))
 
+    @property
+    def num_samples(self):
+        # work round around growing dataset
```
Comment (apaszke) [inline torch/utils/data/sampler.py]: It would be clearer to say that “dataset size might change at runtime”

**Review by fmassa — APPROVED** [review]
Looks good to me thanks!
Could you just address @apaszke comments?

**Comment by truskovskiyk** [comment]
@fmassa Thanks for approving. I already have fixed @apaszke comments.

**Pull request closed without merging**